src/JSON/ZeroCopy/Deserializer.dfy

```diff
@@ -426,19 +517,82 @@ module {:options "-functionSyntax:4"} JSON.ZeroCopy.Deserializer {
           assert SP(arr, cs').StrictlySplitFrom?(cs, Spec.Array);
           Spec.UnfoldValueArray(v);
         }
+        assert sp.StrictlySplitFrom?(cs, Spec.Value);
         Success(sp)
       else if c == '\"' as opt_byte then
-        var SP(str, cs) :- Strings.String(cs);
-        Success(SP(Grammar.String(str), cs))
+        var SP(str, cs') :- Strings.String(cs);
```

Comment:
Worth a comment about why all the expansion/duplication is necessary here?

Reply:
The cases have been there previously. I added the calc's such that post-condition of `Value` goes through. Each of the proofs requires prior knowledge about the shape of `c`, to use either `Grammar.String` or `Grammar.Bool`, and either `var f := _ => TRUE;` or `var f := _ => FALSE;` etc. I think there is not much duplication we can remove here, unfortunately.